src/actions/files.js

```diff
@@ -36,3 +36,48 @@ export function getFilesForPost(teamId, channelId, postId) {
         ]), getState);
     };
 }
+
+const uploadFileRequests = {};
+export function uploadFile(teamId, channelId, fileFormData, formBoundary, rootId, requestId) {
```

Comment:
Yes, the caller generates the requestId since it will be needed to cancel the request. Added this in case the user cancels an upload that would otherwise complete successfully. Since we aren't cancelling the actual upload we use this to check if the request was cancelled before adding the newly uploaded images to the files state.